Remove universal set

mineralntl opened the pull request.

mineralntl committed:
WIP: only setting datatype default to null

mineralntl committed:
Null checks for static methods

mineralntl committed:
Null checks for static methods

mineralntl committed:
Removing more null checks

mineralntl committed:
Removing more null checks

mineralntl committed:
Formatting

mineralntl committed:
Formatting

mineralntl committed:
Updating pom and formatting

mineralntl committed:
Removing breaking get as string method changes

mineralntl commented:
Unit tests will fail until `metadata-utils` is updated

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Updating to new metadata-utils relase

mineralntl commented:
`metadata-utils` has been tagged and updated, everything should be working now

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

hlgp commented on warehouse/query-core/src/main/java/datawave/query/iterator/QueryIterator.java:
```diff
@@ -1417,19 +1416,19 @@ protected NestedIterator<Key> getOrSetKeySource(final Range documentRange, ASTJe
-                                                               // on, don't
-                                                               // attempt
-                                                               // satisfiability
+                    // list is
```
I know these aren't your fault, but I hate the format of these comments

mineralntl replied:
Updated some formatting (hopefully)

hlgp commented on warehouse/query-core/src/test/java/datawave/query/config/ShardQueryConfigurationTest.java:
```diff
@@ -341,7 +341,7 @@ public void testDeepCopyConstructor() {
         Assert.assertEquals(Lists.newArrayList(expectedIndexHole), config.getIndexHoles());
         Assert.assertEquals(Sets.newHashSet("projectFieldA"), config.getProjectFields());
-        Assert.assertEquals(Sets.newHashSet("blacklistedFieldA"), config.getBlacklistedFields());
+        Assert.assertEquals(Sets.newHashSet("blacklistedFieldA"), config.getDisallowedFields());
```
Missed one

mineralntl replied:
Good catch - updated

hlgp commented:
After reading the ticket, I wonder if this should be changed as part of it:
https://github.com/NationalSecurityAgency/datawave/blob/integration/warehouse/query-core/src/main/java/datawave/core/iterators/filter/GlobalIndexDataTypeFilter.java#L42
Seems to violate the second rule unless there's something I'm missing.

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Updating to include global index filter

mineralntl committed:
Missed updating disallowedFields ref

mineralntl committed:
Formatting

jwomeara commented:
Hoping to take a look at this in depth this week.

mineralntl commented:
Good catch @hlgp - updated to match the rules outlined in the ticket

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Updating more refs

mineralntl committed:
Merge branch 'task/RemoveUniversalset-1699' of github.com:NationalSecurityAgency/datawave into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Removing list refs

mineralntl committed:
More ref removal

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

jwomeara commented on warehouse/pom.xml:
```diff
@@ -198,6 +198,11 @@
+            <dependency>
+                <groupId>gov.nsa.datawave.webservices</groupId>
+                <artifactId>metadata-utils</artifactId>
+                <version>${project.version}</version>
```
gov.nsa.datawave.webservices:metadata-utils doesn't exist.  

gov.nsa.datawave.microservice:metadata-utils exists, and it's version is already managed in the parent pom with it's own version, not the project version.

mineralntl replied:
Removed as requested.

jwomeara commented on warehouse/query-core/pom.xml:
```diff
@@ -58,6 +58,7 @@
         <dependency>
             <groupId>gov.nsa.datawave.microservice</groupId>
             <artifactId>metadata-utils</artifactId>
+            <version>1.13</version>
```
We shouldn't use hard-coded versions.  The version of this artifact is already managed in the parent pom, so you can remove the version here.

mineralntl replied:
Understood - removed as requested.

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/QueryParameters.java:
```diff
@@ -43,7 +43,7 @@ public class QueryParameters {
      * ??
      */
-    public static final String BLACKLISTED_FIELDS = "blacklisted.fields";
+    public static final String DISALLOWED_FIELDS = "disallowed.fields";
```
allow/deny?  include/exclude? allow/disallow?  just curious if we're being consistent throughout.

apmoriarty replied:
I'm partial to `include`/`exclude` when updating our language. Just my two cents.

mineralntl replied:
Removing these changes as updating terms should probably be its own effort

jwomeara reviewed (changes requested):
This is a good start, but there are some additional changes that I was hoping to see made with this ticket.  There are multiple places in the code where we pass a list of types to filter, but we end up just ignoring that list of types.  Some places I've seen this are:
FacetedVisitor
GeoWaveFunctionsDescriptor
GeoFunctionsDescriptor
EvaluationPhaseFilterFunctionsDescriptor
GroupingRequiredFilterFunctionsDescriptor
QueryFunctionsDescriptor

There are also places where this list of fields ends up being used, but the functionality might differ now that null is used in place of a universal set isn't used.  I've seen this in:
DateIndexHelper

Everything else looks fine.

jwomeara commented:
Oh also, WiredQueryExecutorBeanTest failed for me when I built this.  You can run this test as part of your build with -Ddeploy

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Removing as unnecessary per team feedback

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'task/RemoveUniversalset-1699' of github.com:NationalSecurityAgency/datawave into task/RemoveUniversalset-1699

mineralntl committed:
Updating broken method params list

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Missed semicolon during merge fix

mineralntl committed:
Updating DateIndexHelper to account for empty typeFilters

mineralntl committed:
Formatting

mineralntl commented:
Changing to draft as various classes are updated in pieces

mineralntl commented:
Adding FieldMissingFromSchemaVisitor/Test as well - found during testing.

mineralntl committed:
Updating several field calls - some tests still broken

mineralntl committed:
Minor tweaks - still not working

mineralntl committed:
Updating function handling

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Reverting term updates to its own effort

mineralntl committed:
Reverting more term updates to their own effort

mineralntl committed:
Reverting more term updates to their own effort again

mineralntl committed:
Fixing unit test field typo

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl commented:
These changes are tied to [datawave-metadata-utils#20](https://github.com/NationalSecurityAgency/datawave-metadata-utils/pull/20) as this branch is dependent on those changes for fully removing UniversalSet and handling empty but non-null datatype filters.

mineralntl committed:
Adding GroupingRequiredDescriptor

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
More updates to type filters

mineralntl committed:
Updating thrown exceptions

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/functions/GroupingRequiredFilterFunctionsDescriptor.java:
```diff
@@ -110,28 +115,55 @@ public Set<String> fields(MetadataHelper helper, Set<String> datatypeFilter) {
             }
-            return fields;
+            
+            for (String field : fields) {
+                filterField(allFields, field, filteredFields);
+            }
```
I would update filterField to accept the set of fields, and remove the for loop here since that's how it's being used throughout this class.

mineralntl replied:
Updated

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/functions/GroupingRequiredFilterFunctionsDescriptor.java:
```diff
@@ -110,28 +115,55 @@ public Set<String> fields(MetadataHelper helper, Set<String> datatypeFilter) {
+            
+            for (Set<String> aFieldSet : fieldSets) {
+                Set<String> filteredFields = Sets.newHashSet();
+                for (String field : aFieldSet) {
+                    filterField(allFields, field, filteredFields);
+                }
```
Same comment here as above.

mineralntl replied:
Updated

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/functions/QueryFunctionsDescriptor.java:
```diff
@@ -92,7 +95,7 @@ private JexlNode getIndexQuery() {
         }
         
-        private JexlNode getTextIndexQuery() {
+        private JexlNode getTextIndexQuery(MetadataHelper helper, Set<String> datatypeFilter) {
```
You added the params here, but you aren't using them.  Should we?

mineralntl replied:
Changed to calculating the fields before sending to method

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/functions/QueryFunctionsDescriptor.java:
```diff
@@ -61,17 +64,17 @@ public JexlNode getIndexQuery(ShardQueryConfiguration config, MetadataHelper hel
         }
         
-        private JexlNode getIndexQuery() {
+        private JexlNode getIndexQuery(MetadataHelper helper, Set<String> datatypeFilter) throws TableNotFoundException {
```
You added the params here, but you aren't using them. Should we?

mineralntl replied:
Changed to calculating the fields before sending to method

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/functions/QueryFunctionsDescriptor.java:
```diff
@@ -61,17 +64,17 @@ public JexlNode getIndexQuery(ShardQueryConfiguration config, MetadataHelper hel
                     // Return an index query.
-                    return getIndexQuery();
+                    return getIndexQuery(helper, datatypeFilter);
                 case INCLUDE_TEXT:
                     // Return the appropriate index query.
-                    return getTextIndexQuery();
+                    return getTextIndexQuery(helper, datatypeFilter);
```
Maybe instead of passing down the metadata helper and the data type filter, you could run the field filtering prior to these calls and just pass the resulting fields down.

mineralntl replied:
Changed to calculating the fields before sending to method

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/CaseSensitivityVisitor.java:
```diff
@@ -62,7 +63,12 @@ public Object visit(ASTFunctionNode node, Object data) {
         
-        Set<String> fields = desc.fields(helper, config.getDatatypeFilter());
+        Set<String> fields = null;
+        try {
+            fields = desc.fields(helper, config.getDatatypeFilter());
+        } catch (TableNotFoundException | InstantiationException | IllegalAccessException e) {
+            throw new RuntimeException(e);
```
Can we add a log message here as well?

mineralntl replied:
Added

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/EventDataQueryExpressionVisitor.java:
```diff
@@ -422,7 +434,11 @@ public Object visit(ASTAndNode node, Object data) {
+        try {
+            desc.addFilters(attributeFactory, filterMap);
+        } catch (TableNotFoundException | InstantiationException | IllegalAccessException e) {
+            throw new RuntimeException(e);
```
Can we add a log message here as well?

mineralntl replied:
Added

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExpandMultiNormalizedTerms.java:
```diff
@@ -140,7 +138,11 @@ public Object visit(ASTGENode node, Object data) {
+        try {
+            return FunctionNormalizationRebuildingVisitor.normalize(node, config.getQueryFieldsDatatypes(), helper, config.getDatatypeFilter());
+        } catch (TableNotFoundException e) {
+            throw new RuntimeException(e);
```
Same comment here about a log message.

mineralntl replied:
Added

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/FetchDataTypesVisitor.java:
```diff
@@ -189,38 +186,42 @@ public Object visit(ASTLTNode node, Object data) {
+                }
             }
+        } catch (TableNotFoundException | InstantiationException | IllegalAccessException e) {
+            throw new RuntimeException(e);
```
Same comment about logging the error.  Starting to wonder whether it would have made sense to handle the exceptions within the method if all we're ever going to do is wrap it in a runtime exception.

mineralntl replied:
Updated to handle known exceptions with better logging - will also update the rest of the visitors with the same logic.

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/FieldMissingFromSchemaVisitor.java:
```diff
@@ -148,13 +148,20 @@ public Object visit(ASTFunctionNode node, Object data) {
+                genericVisit(node, data);
             }
+        } catch (TableNotFoundException | InstantiationException | IllegalAccessException e) {
+            throw new RuntimeException(e);
```
same comment about logging.

mineralntl replied:
Same comment from FetchDataTypesVisitor about updating exception handling and logging logic

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/FunctionIndexQueryExpansionVisitor.java:
```diff
@@ -114,7 +114,14 @@ public Object visit(ASTFunctionNode node, Object data) {
+                    indexQuery = desc.getIndexQuery(config, this.metadataHelper, this.dateIndexHelper, this.config.getDatatypeFilter());
+                }
+            } catch (TableNotFoundException e) {
+                throw new RuntimeException(e);
```
logging.

mineralntl replied:
Logged.

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/FunctionIndexQueryExpansionVisitor.java:
```diff
@@ -114,7 +114,14 @@ public Object visit(ASTFunctionNode node, Object data) {
-            JexlNode indexQuery = desc.getIndexQuery(config, this.metadataHelper, this.dateIndexHelper, this.config.getDatatypeFilter());
+            JexlNode indexQuery = null;
+            try {
+                if (desc != null) {
```
Were you seeing cases where desc was null, or is this just precautionary?  I might consider moving this up to the prior if statement.

mineralntl replied:
Was just precautionary - will move as suggested

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PushFunctionsIntoExceededValueRanges.java:
```diff
@@ -86,15 +86,19 @@ public Object visit(ASTAndNode node, Object data) {
-            } else {
-                children.add(child);
+            } catch (TableNotFoundException | InstantiationException | IllegalAccessException e) {
+                throw new RuntimeException(e);
```
logging.

mineralntl replied:
Logged.

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/QueryFieldsVisitor.java:
```diff
@@ -156,7 +157,12 @@ public Object visit(ASTIdentifier node, Object data) {
-        Set<String> fields = desc.fields(helper, null);
+        Set<String> fields = null;
+        try {
+            fields = desc.fields(helper, null);
```
no null check for desc here?  just asking since i saw you add one in another class.

mineralntl replied:
Whoops - good catch, that's an oversight.

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/QueryFieldsVisitor.java:
```diff
@@ -156,7 +157,12 @@ public Object visit(ASTIdentifier node, Object data) {
+        } catch (TableNotFoundException | InstantiationException | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
         ((Set<String>) data).addAll(fields);
```
if fields is null, this is going to throw a NPE.  maybe move addAll into the try block?

mineralntl replied:
Another oversight - good catch.

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/whindex/SplitGeoWaveFunctionVisitor.java:
```diff
@@ -37,7 +38,12 @@ public static <T extends JexlNode> T apply(T script, MetadataHelper metadataHelp
+            try {
+                fields = descriptor.fields(metadataHelper, Collections.emptySet());
+            } catch (TableNotFoundException | InstantiationException | IllegalAccessException e) {
+                throw new RuntimeException(e);
```
i am really thinking we should handle these exceptions in the method...

mineralntl replied:
Handled in latest update

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/whindex/SplitGeoWaveFunctionVisitor.java:
```diff
@@ -60,7 +66,12 @@ public Object visit(ASTFunctionNode node, Object data) {
+            try {
+                fields = descriptor.fields(metadataHelper, Collections.emptySet());
+            } catch (TableNotFoundException | InstantiationException | IllegalAccessException e) {
+                throw new RuntimeException(e);
```
logging.

mineralntl replied:
Handled better in latest update

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/whindex/SplitGeoWaveFunctionVisitor.java:
```diff
@@ -37,7 +38,12 @@ public static <T extends JexlNode> T apply(T script, MetadataHelper metadataHelp
+            } catch (TableNotFoundException | InstantiationException | IllegalAccessException e) {
+                throw new RuntimeException(e);
+            }
             if (fields.size() > 1) {
```
If fields can be null, this could be a problem.

mineralntl replied:
Initialized empty but non-null set

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/whindex/SplitGeoWaveFunctionVisitor.java:
```diff
@@ -60,7 +66,12 @@ public Object visit(ASTFunctionNode node, Object data) {
+            } catch (TableNotFoundException | InstantiationException | IllegalAccessException e) {
+                throw new RuntimeException(e);
+            }
             if (fields.size() > 1) {
```
if fields can be null, this might be a problem.

mineralntl replied:
Initialized empty but non-null set

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/whindex/WhindexVisitor.java:
```diff
@@ -302,7 +302,12 @@ public Object visit(ASTAndNode node, Object data) {
+        try {
+            leafNodes = getLeafNodes(node, nonLeafNodes);
+        } catch (TableNotFoundException | InstantiationException | IllegalAccessException e) {
+            throw new RuntimeException(e);
```
logging.

mineralntl replied:
Handled better in updated commit

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/tld/TLDQueryIterator.java:
```diff
@@ -117,8 +117,8 @@ public EventDataQueryFilter getEvaluationFilter() {
             this.evaluationFilter = new TLDEventDataFilter(script, getAllFields(), typeMetadata, useWhiteListedFields ? whiteListedFields : null,
-                            useBlackListedFields ? blackListedFields : null, getEventFieldSeek(), getEventNextSeek(),
-                            limitFieldsPreQueryEvaluation ? limitFieldsMap : Collections.emptyMap(), limitFieldsField, getNonEventFields());
+                            useBlackListedFields ? blackListedFields : null, maxFieldHitsBeforeSeek, maxKeysBeforeSeek,
```
Why was this changed?

mineralntl replied:
99% sure this was a fat fingered mistake from a botched merge attempt. Reverting to match `integration`.

jwomeara commented on warehouse/query-core/src/test/java/datawave/query/config/ShardQueryConfigurationTest.java:
```diff
@@ -328,7 +328,7 @@ public void testDeepCopyConstructor() {
         indexHoles.add(otherIndexHole);
         projectFields.add("projectFieldB");
-        blacklistedFields.add("blacklistedFieldB");
+        blacklistedFields.add("blacklistedfieldB");
```
is the case correct?

mineralntl replied:
Fat fingered from a black/white list revert attempt. Reverting the revert. Good catch.

jwomeara reviewed (changes requested):
I appreciate your work on this.  Sorry for the delay w.r.t. the review.  Just a few comments.

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Updates for team feedback

mineralntl committed:
Undoing botched merge attempt

mineralntl committed:
More fat fingered reverts

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Missed merge diff

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Calculate fields before sending to getIndexQuery

mineralntl committed:
Support for Geo descriptor updates

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Updates for new baseline version

mineralntl committed:
Updating exceptions

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Including geo/wave descriptors in filtering null/empty type filters

jwomeara commented:
Is this ready to be reviewed again?

mineralntl committed:
Update pom.xml for metadata utils release

jwomeara committed:
Removed unnecessary exceptions

mineralntl committed:
Updating logging for exceptions

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl commented:
All comments addressed - should be good to go

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/GeoWavePruningVisitor.java:
```diff
@@ -76,19 +79,21 @@ public Object visit(ASTAndNode node, Object data) {
-                            } catch (TableNotFoundException | ExecutionException | MarkingFunctions.Exception e) {
-                                log.debug("Could not load datatypes from metadata table");
-                            }
+                            fields = geoWaveDesc.fields(metadataHelper, null);
```
This could be condensed into a single line.

mineralntl replied:
Done and done.

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/whindex/SplitGeoWaveFunctionVisitor.java:
```diff
@@ -41,13 +45,22 @@ public static <T extends JexlNode> T apply(T script, MetadataHelper metadataHelp
     @Override
     public Object visit(ASTFunctionNode node, Object data) {
         JexlArgumentDescriptor descriptor = JexlFunctionArgumentDescriptorFactory.F.getArgumentDescriptor(node);
+        Set<String> fields = Sets.newHashSet();
```
Just curious...  Is there an advantage to doing this vs. `new HashSet<>()` ?  Also, in this case specifically, we never use the set created by this call, so we should probably just assign null.

mineralntl replied:
There wasn't a particular advantage I was going for - just following what I had already seen in other places throughout the codebase - but also, I'll go ahead and remove the empty set.

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/whindex/SplitGeoWaveFunctionVisitor.java:
```diff
@@ -41,13 +45,22 @@ public static <T extends JexlNode> T apply(T script, MetadataHelper metadataHelp
-            } catch (TableNotFoundException | InstantiationException | IllegalAccessException | ExecutionException |
-                     MarkingFunctions.Exception e) {
-                LOGGER.debug("Unable to load datatypes from metadata table");
+            } catch (TableNotFoundException e) {
+                QueryException qe = new QueryException(DatawaveErrorCode.METADATA_TABLE_FETCH_ERROR, e);
+                LOGGER.error(qe);
+                throw new DatawaveFatalQueryException(qe);
+            } catch (InstantiationException | IllegalAccessException e) {
+                QueryException qe = new QueryException(DatawaveErrorCode.METADATA_TABLE_RECORD_FETCH_ERROR, e);
+                LOGGER.error(qe);
+                throw new DatawaveFatalQueryException(qe);
+            } catch (ExecutionException | MarkingFunctions.Exception e) {
+                QueryException qe = new QueryException(DatawaveErrorCode.UNKNOWN_SERVER_ERROR, e);
+                LOGGER.error(qe);
+                throw new DatawaveFatalQueryException(qe);
```
I think I made this comment before, but why not put this exception handling in the `fields` method?  It seems like we're copying the same exception handling all over the place when we could probably just consolidate it by having it taken care of within the method.

mineralntl replied:
Ah, I misunderstood the original comment - will update fields() and similar methods to remove the repeating exception handling.

jwomeara reviewed (changes requested):
see comments.

mineralntl committed:
Updating based on team feedback

mineralntl committed:
More cleanup of exception handling

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/index/lookup/ShardRangeStream.java:
```diff
@@ -29,55 +26,58 @@
+import java.util.Date;
+import java.util.Iterator;
+import java.util.Map.Entry;
+import java.util.concurrent.ExecutionException;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/index/lookup/ShardRangeStream.java:
```diff
@@ -1,14 +1,12 @@
-import datawave.query.index.lookup.IndexStream.StreamContext;
+import com.google.common.base.Function;
+import com.google.common.collect.Iterators;
+import datawave.marking.MarkingFunctions;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java:
```diff
@@ -1,15 +1,10 @@
+import com.google.common.collect.Lists;
+import com.google.common.collect.PeekingIterator;
+import com.google.common.collect.Sets;
+import datawave.marking.MarkingFunctions;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java:
```diff
@@ -50,6 +39,17 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/AllTermsIndexedVisitor.java:
```diff
@@ -1,6 +1,7 @@
 package datawave.query.jexl.visitors;
 
 import com.google.common.base.Preconditions;
+import datawave.marking.MarkingFunctions;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/AllTermsIndexedVisitor.java:
```diff
@@ -31,6 +32,7 @@
 
 import java.text.MessageFormat;
 import java.util.NoSuchElementException;
+import java.util.concurrent.ExecutionException;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/BoundedRangeDetectionVisitor.java:
```diff
@@ -1,5 +1,6 @@
 package datawave.query.jexl.visitors;
 
+import datawave.marking.MarkingFunctions;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/BoundedRangeDetectionVisitor.java:
```diff
@@ -18,28 +19,29 @@
 import org.apache.commons.jexl2.parser.ASTReference;
 import org.apache.commons.jexl2.parser.JexlNode;
 
+import java.util.concurrent.ExecutionException;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExecutableDeterminationVisitor.java:
```diff
@@ -2,6 +2,7 @@
 
 import com.google.common.collect.HashMultimap;
 import com.google.common.collect.Multimap;
+import datawave.marking.MarkingFunctions;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExecutableDeterminationVisitor.java:
```diff
@@ -77,6 +78,7 @@
 import java.util.List;
 import java.util.NoSuchElementException;
 import java.util.Set;
+import java.util.concurrent.ExecutionException;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PullupUnexecutableNodesVisitor.java:
```diff
@@ -1,5 +1,6 @@
 package datawave.query.jexl.visitors;
 
+import datawave.marking.MarkingFunctions;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PullupUnexecutableNodesVisitor.java:
```diff
@@ -62,6 +63,7 @@
 import org.apache.log4j.Logger;
 
 import java.util.Set;
+import java.util.concurrent.ExecutionException;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PushdownMissingIndexRangeNodesVisitor.java:
```diff
@@ -1,5 +1,6 @@
 package datawave.query.jexl.visitors;
 
+import datawave.marking.MarkingFunctions;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PushdownMissingIndexRangeNodesVisitor.java:
```diff
@@ -24,14 +25,15 @@
 import java.util.Set;
 import java.util.SortedSet;
 import java.util.TreeSet;
+import java.util.concurrent.ExecutionException;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PushdownUnexecutableNodesVisitor.java:
```diff
@@ -1,5 +1,6 @@
 package datawave.query.jexl.visitors;
 
+import datawave.marking.MarkingFunctions;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PushdownUnexecutableNodesVisitor.java:
```diff
@@ -61,22 +62,23 @@
 import org.apache.log4j.Logger;
 
 import java.util.Set;
+import java.util.concurrent.ExecutionException;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/planner/MetadataHelperQueryModelProvider.java:
```diff
@@ -1,26 +1,29 @@
 package datawave.query.planner;
 
+import datawave.marking.MarkingFunctions;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/planner/MetadataHelperQueryModelProvider.java:
```diff
@@ -1,26 +1,29 @@
 import org.apache.accumulo.core.client.TableNotFoundException;
 import org.apache.log4j.Logger;
 
+import java.util.concurrent.ExecutionException;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/planner/pushdown/CostEstimator.java:
```diff
@@ -1,67 +1,68 @@
-
-import java.util.NoSuchElementException;
-
+import datawave.marking.MarkingFunctions;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/planner/pushdown/CostEstimator.java:
```diff
@@ -1,67 +1,68 @@
 import org.apache.log4j.Logger;
 
+import java.util.NoSuchElementException;
+import java.util.concurrent.ExecutionException;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/planner/rules/RegexPushdownTransformRule.java:
```diff
@@ -1,26 +1,27 @@
 package datawave.query.planner.rules;
 
+import datawave.marking.MarkingFunctions;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/planner/rules/RegexPushdownTransformRule.java:
```diff
@@ -1,26 +1,27 @@
 import org.apache.log4j.Logger;
 
 import java.util.List;
+import java.util.concurrent.ExecutionException;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/tables/async/event/VisitorFunction.java:
```diff
@@ -4,6 +4,7 @@
 import com.google.common.cache.Cache;
 import com.google.common.cache.CacheBuilder;
 import datawave.core.iterators.filesystem.FileSystemCache;
+import datawave.marking.MarkingFunctions;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/main/java/datawave/query/tables/async/event/VisitorFunction.java:
```diff
@@ -57,37 +58,38 @@
 import java.util.Set;
 import java.util.SortedMap;
 import java.util.TreeMap;
+import java.util.concurrent.ExecutionException;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/test/java/datawave/query/jexl/functions/GeoFunctionsDescriptorTest.java:
```diff
@@ -5,8 +5,10 @@
 import datawave.query.jexl.JexlASTHelper;
 import datawave.query.jexl.functions.arguments.JexlArgumentDescriptor;
 import datawave.query.jexl.visitors.JexlStringBuildingVisitor;
+import org.apache.accumulo.core.client.TableNotFoundException;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/test/java/datawave/query/jexl/functions/GeoFunctionsDescriptorTest.java:
```diff
@@ -5,8 +5,10 @@
+import org.apache.accumulo.core.client.TableNotFoundException;
 import org.apache.commons.jexl2.parser.ASTFunctionNode;
 import org.apache.commons.jexl2.parser.JexlNode;
+import org.apache.commons.jexl2.parser.ParseException;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/test/java/datawave/query/jexl/visitors/PullupUnexecutableNodesVisitorTest.java:
```diff
@@ -1,159 +1,162 @@
+
     @Before
-    public void setup() {
+    public void setup() throws TableNotFoundException, ExecutionException, MarkingFunctions.Exception {
```
unnecessary.  and imports unneeded after removed.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/test/java/datawave/query/jexl/visitors/PushdownUnexecutableNodesVisitorTest.java:
```diff
@@ -233,58 +235,58 @@ public PushdownUnexecutableNodesVisitorTest(String testName, String baseQuery, E
+
     @Before
-    public void setup() throws TableNotFoundException {
+    public void setup() throws TableNotFoundException, ExecutionException, MarkingFunctions.Exception {
```
last two exceptions not needed.  imports not needed either.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/test/java/datawave/query/jexl/visitors/QueryFieldsVisitorTest.java:
```diff
@@ -2,206 +2,214 @@
+
     @BeforeClass
-    public static void setup() {
+    public static void setup() throws TableNotFoundException, ExecutionException, MarkingFunctions.Exception {
```
not needed, and imports can be removed.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/test/java/datawave/query/jexl/visitors/GeoWavePruningVisitorTest.java:
```diff
@@ -14,122 +16,124 @@
+
     @BeforeClass
-    public static void setup() {
+    public static void setup() throws TableNotFoundException, ExecutionException, MarkingFunctions.Exception {
```
not needed.  remove imports.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/test/java/datawave/query/jexl/visitors/QueryModelVisitorTest.java:
```diff
@@ -299,90 +301,90 @@ public void testMoreMethods() throws ParseException {
+
     @Test
-    public void testAppliedModelWithNullNoFail() throws ParseException {
+    public void testAppliedModelWithNullNoFail() throws ParseException, TableNotFoundException, ExecutionException, MarkingFunctions.Exception {
```
last three exceptions unnecessary.  remove imports as well.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/test/java/datawave/query/jexl/visitors/UnfieldedIndexExpansionVisitorTest.java:
```diff
@@ -37,161 +37,162 @@
+
     @Before
-    public void setupTest() {
+    public void setupTest() throws TableNotFoundException, ExecutionException, MarkingFunctions.Exception {
```
exceptions unnecessary.  remove imports.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/test/java/datawave/query/tables/async/event/VisitorFunctionTest.java:
```diff
@@ -74,52 +76,52 @@ private void setupExpects() throws TableNotFoundException, IOException, URISynta
+
     @Test
-    public void underTermThresholdTest() throws IOException, TableNotFoundException, URISyntaxException {
+    public void underTermThresholdTest() throws IOException, TableNotFoundException, URISyntaxException, ExecutionException, MarkingFunctions.Exception {
```
last two exceptions unnecessary.  remove imports.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/test/java/datawave/query/tables/facets/FacetCheckTest.java:
```diff
@@ -28,211 +30,211 @@
     @Before
-    public void before() throws TableNotFoundException, IllegalAccessException, InstantiationException {
-        Multimap<String,String> facets = HashMultimap.create();
+    public void before() throws TableNotFoundException, IllegalAccessException, InstantiationException, ExecutionException, MarkingFunctions.Exception {
```
last two exceptions unnecessary.  remove imports.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/test/java/datawave/query/util/MockMetadataHelper.java:
```diff
@@ -37,64 +36,65 @@
 import java.util.Set;
 import java.util.TimeZone;
 import java.util.concurrent.ExecutionException;
```
unused.

mineralntl replied:
Removed

jwomeara commented on warehouse/query-core/src/test/java/datawave/query/util/MockMetadataHelper.java:
```diff
@@ -18,7 +18,6 @@
 import datawave.query.model.QueryModel;
 import datawave.util.TableName;
 import org.apache.accumulo.core.client.AccumuloClient;
-import org.apache.accumulo.core.client.AccumuloException;
```
MarkingFunctions import above is unused.

mineralntl replied:
Removed

jwomeara reviewed (changes requested):
unused import and exception cleanup.  this is 99% of the way there.  let's make these final changes, wrap it up, and finally be done with it.

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Removing more unsed imports

jwomeara committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Removing unnecessary whitespace changes

jwomeara commented on warehouse/ingest-configuration/src/main/resources/config/edge-definitions.xml:
```diff
@@ -1220,7 +1220,7 @@ http://www.springframework.org/schema/util/spring-util-4.0.xsd">
     </bean>
 
-    <bean id="default.protobufedge.table.blacklist.values" class="org.springframework.beans.factory.config.SetFactoryBean">
+    <bean id="default.protobufedge.table.disallowlist.values" class="org.springframework.beans.factory.config.SetFactoryBean">
```
I thought we were saving these changes for your other PR?  This was good to go, but this commit dismissed all of the approvals...  Also, isn't this change meaningless without also changing EDGE_TABLE_BLACKLIST_VALUES in ProtobufEdgeDataTypeHandler?

mineralntl replied:
This is me confusing two different PRs and now needing to revert the changes made - totally unintentional.

mineralntl committed:
Revert "Removing unnecessary whitespace changes"

This reverts commit 60db51f1598ab0b0bfad8a6cc1e2b8171b985158.

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Formatting due to downstream dependency merge conflict

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

jwomeara reviewed (changes requested):
Formatting still doesn't look right.  Also, it seems you undid @apmoriarty 's batched query pr.

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

# Conflicts:
#	warehouse/query-core/src/main/java/datawave/core/iterators/filter/GlobalIndexDataTypeFilter.java
#	warehouse/query-core/src/main/java/datawave/query/discovery/DiscoveryLogic.java
#	warehouse/query-core/src/main/java/datawave/query/index/lookup/RangeStream.java
#	warehouse/query-core/src/main/java/datawave/query/index/lookup/ShardRangeStream.java
#	warehouse/query-core/src/main/java/datawave/query/iterator/QueryIterator.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/JexlASTHelper.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/functions/ContentFunctionsDescriptor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/functions/EvaluationPhaseFilterFunctionsDescriptor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/functions/GeoFunctionsDescriptor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/functions/GeoWaveFunctionsDescriptor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/functions/GroupingRequiredFilterFunctionsDescriptor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/functions/QueryFunctionsDescriptor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/functions/arguments/JexlArgumentDescriptor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/lookups/RegexIndexLookup.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/lookups/ShardIndexQueryTableStaticMethods.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/AllTermsIndexedVisitor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/BaseIndexExpansionVisitor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/BoundedRangeIndexExpansionVisitor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/CaseSensitivityVisitor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/EventDataQueryExpressionVisitor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExecutableDeterminationVisitor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExpandMultiNormalizedTerms.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/FetchDataTypesVisitor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/FieldMissingFromSchemaVisitor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/FunctionIndexQueryExpansionVisitor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/FunctionNormalizationRebuildingVisitor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/GeoWavePruningVisitor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PullupUnexecutableNodesVisitor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PushFunctionsIntoExceededValueRanges.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PushdownMissingIndexRangeNodesVisitor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PushdownUnexecutableNodesVisitor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/QueryFieldsVisitor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexIndexExpansionVisitor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/UnfieldedIndexExpansionVisitor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/whindex/SplitGeoWaveFunctionVisitor.java
#	warehouse/query-core/src/main/java/datawave/query/jexl/visitors/whindex/WhindexVisitor.java
#	warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java
#	warehouse/query-core/src/main/java/datawave/query/planner/pushdown/CostEstimator.java
#	warehouse/query-core/src/main/java/datawave/query/predicate/TLDEventDataFilter.java
#	warehouse/query-core/src/main/java/datawave/query/tables/ShardQueryLogic.java
#	warehouse/query-core/src/main/java/datawave/query/tables/async/event/VisitorFunction.java
#	warehouse/query-core/src/main/java/datawave/query/tables/facets/FacetedVisitor.java
#	warehouse/query-core/src/main/java/datawave/query/util/DateIndexHelper.java
#	warehouse/query-core/src/test/java/datawave/query/index/lookup/RangeStreamQueryTest.java
#	warehouse/query-core/src/test/java/datawave/query/jexl/functions/ContentFunctionsTest.java
#	warehouse/query-core/src/test/java/datawave/query/jexl/functions/GeoWaveFunctionsDescriptorTest.java
#	warehouse/query-core/src/test/java/datawave/query/jexl/visitors/ExecutableDeterminationVisitorTest.java
#	warehouse/query-core/src/test/java/datawave/query/jexl/visitors/ExpandCompositeTermsTest.java
#	warehouse/query-core/src/test/java/datawave/query/jexl/visitors/ExpandMultiNormalizedTermsTest.java
#	warehouse/query-core/src/test/java/datawave/query/jexl/visitors/FieldMissingFromSchemaVisitorTest.java
#	warehouse/query-core/src/test/java/datawave/query/jexl/visitors/GeoWavePruningVisitorTest.java
#	warehouse/query-core/src/test/java/datawave/query/jexl/visitors/PullupUnexecutableNodesVisitorTest.java
#	warehouse/query-core/src/test/java/datawave/query/jexl/visitors/PushdownUnexecutableNodesVisitorTest.java
#	warehouse/query-core/src/test/java/datawave/query/jexl/visitors/QueryModelVisitorTest.java
#	warehouse/query-core/src/test/java/datawave/query/jexl/visitors/UnfieldedIndexExpansionVisitorTest.java
#	warehouse/query-core/src/test/java/datawave/query/jexl/visitors/WhindexVisitorTest.java
#	warehouse/query-core/src/test/java/datawave/query/predicate/TLDEventDataFilterTest.java
#	warehouse/query-core/src/test/java/datawave/query/util/DateIndexHelperTest.java
#	warehouse/query-core/src/test/java/datawave/query/util/MockMetadataHelper.java

mineralntl committed:
Testing formatter changes for ease of PR reviews

mineralntl committed:
More formatting tests

mineralntl committed:
Matching integration formatting

mineralntl committed:
Undoing weird merge conflict error

mineralntl committed:
Re-adding missing changes during merge

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl commented:
Somehow removed a bunch of changes during last attempt at fixing the merge conflicts - reverted and attempting again

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Formatting

mineralntl committed:
Fixing formatting and sorting

mineralntl committed:
More formatting weirdness

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Updating based on downstream testing

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Updating to include newest metadata-utils testing jar

hlgp committed:
Update tests.yml to force update

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Updating to newer metadata-utils

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Updating based on large codebase changes

mineralntl committed:
Updating unit tests to removal empty set

mineralntl committed:
Another unit test update

mineralntl committed:
Formatting

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Adding null checks

mineralntl committed:
Checking for previously populated type list

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Updating microservices

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

hlgp committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

hlgp commented:
Looking at this for the first time in a while.  If I heard correctly, there was some disagreement about whether null or empty should signify all datatypes.  @apmoriarty  @mineralntl  correct me if I'm wrong.   Either seems a bit fragile.  If the purpose of this was to prevent an incidental return of all datatypes because someone unwittingly set it to be a new set somewhere (and hence why the original suggestion was null) maybe it's safer to come up with a DataTypeFilter object with an enum of ALL, NONE, and SET along with the Set?  Sorry I know that'd be a lot more work.  What do you guys think?

apmoriarty commented:
> Looking at this for the first time in a while. If I heard correctly, there was some disagreement about whether null or empty should signify all datatypes. @apmoriarty @mineralntl correct me if I'm wrong. Either seems a bit fragile. If the purpose of this was to prevent an incidental return of all datatypes because someone unwittingly set it to be a new set somewhere (and hence why the original suggestion was null) maybe it's safer to come up with a DataTypeFilter object with an enum of ALL, NONE, and SET along with the Set? Sorry I know that'd be a lot more work. What do you guys think?

In places where we do field filtering it's not uncommon to see something like this:
```java
public boolean accepted(String field) {
    return includeFields == null || includeFields.contains(field);
}
```
It seems reasonable to push the filtering logic into a helper class to ensure consistent behavior throughout the code base. I could see a filter factory that enforces "allow some" or "allow all" modes. I can't think of any situation where we want to exclude all datatypes.

hlgp commented:
> It seems reasonable to push the filtering logic into a helper class to ensure consistent behavior throughout the code base. I could see a filter factory that enforces "allow some" or "allow all" modes. I can't think of any situation where we want to exclude all datatypes.

Helper/factory sounds good.  I guess a user could supply a datatype that doesn't exist, but I'd hope we'd short circuit 204 in that case.

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Updating with TypeFilter enum

mineralntl committed:
Updating more null/empty references

mineralntl committed:
Updating unit tests

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Adding md-utils support

mineralntl committed:
Adding TypeFilter class

mineralntl commented:
Moved metadata-utils support classes and changes to main repo; should no longer require external support to build and compile correctly

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

ivakegg committed:
4.0.12

ivakegg committed:
4.0.13-SNAPSHOT

avgAGB committed:
Formatting integration (#2800)

ivakegg committed:
Updated the metadata helper to use the correct table name for (#2801)

* Updated the metadata helper to use the correct table name for getAllFields when getting the query model
* Added hashCode to the UniversalSet to allow for appropriate cache keys

ivakegg committed:
4.0.13

ivakegg committed:
4.0.14-SNAPSHOT

ivakegg committed:
Ensure ingestTypeFilter gets passed through

ivakegg committed:
4.0.14

ivakegg committed:
4.0.15-SNAPSHOT

avgAGB committed:
RegexIndexExpansion cannot catch unthrown exception (#2805)

Co-authored-by: Ivan Bella <347158+ivakegg@users.noreply.github.com>
Co-authored-by: Keith Ratcliffe <ratcliffe.keith@gmail.com>

mineralntl committed:
Spelling/formatting

avgAGB committed:
Converting strs to chars on lastIndexOf (#2782)

* Converting strs to chars on lastIndexOf

* Removing test annotation

* Disabled for Junit5 has replaced ignore

* Formatting for MetadataHelper

---------

Co-authored-by: Adam Lerman <awlerma@uwe.nsa.gov>

drewfarris committed:
Decompose identifiers prior to looking up field normalizers for content functions (#2802)

* Fields names (identifiers) starting with numbers require a leading '$' in JEXL (e.g., $1234_A)
* Datatypes/normalizers for these fields are registered te field without the leading '$' (e.g 1234_A)
* When looking up the normalizers to apply to a field in the context of content functions, the '$' was not being removed from the field names, so the proper normalizers weren't found.

This fix add the call to properly clean the field name before attempting to look up the normalizer.

apmoriarty committed:
Address race condition where expansion is successful but more values … (#2797)

* Address race condition where expansion is successful but more values are added before the result is logged

* Wrap whole block

austin007008 committed:
rename field CONTENT_SUMMARY to SUMMARY (#2795)

* rename field CONTENT_SUMMARY to SUMMARY

* fix getter/setter name in ShardQueryLogic

mineralntl committed:
Updating pom for metadata-utils

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

mineralntl committed:
Changing FieldMissing

mineralntl committed:
Merge branch 'integration' into task/RemoveUniversalset-1699

hlgp commented:
OBE
